x/txfees/keeper/fees.go

```diff
@@ -44,43 +47,38 @@ func (k Keeper) ChargeFeesFromPayer(
 // If the fee token is unknown, it is sent to the community pool.
 func (k Keeper) ChargeFees(
 	ctx sdk.Context,
-	takerFeeCoin sdk.Coin,
+	takerFee sdk.Coin,
 	beneficiary *sdk.AccAddress,
 	payer string, // optional, only used for the event
 ) error {
-	if takerFeeCoin.IsZero() {
+	if takerFee.IsZero() {
 		// Nothing to charge
 		return nil
 	}
 
 	// Swap the taker fee to the base denom
-	baseDenomFee, communityPoolCoins, err := k.swapFeeToBaseDenom(ctx, takerFeeCoin)
+	baseDenomFee, communityPool, err := k.swapFeeToBaseDenom(ctx, takerFee)
 	if err != nil {
 		return fmt.Errorf("swap fee to base denom: %w", err)
 	}
 
 	// If the fee token is unknown or the swap is unsuccessful, the fee is sent to the community pool.
-	if !communityPoolCoins.Empty() {
+	if !communityPool.Empty() {
 		// Send unknown fee tokens to the community pool
-		err = k.communityPool.FundCommunityPool(ctx, communityPoolCoins, k.accountKeeper.GetModuleAddress(types.ModuleName))
+		err = k.communityPool.FundCommunityPool(ctx, communityPool, k.accountKeeper.GetModuleAddress(types.ModuleName))
 		if err != nil {
-			return fmt.Errorf("unknown fee token: func community pool: %w", err)
+			return fmt.Errorf("fund community pool: %w", err)
 		}
 
-		k.Logger(ctx).With("fee", communityPoolCoins.String(), "error", err).
-			Error("Cannot swap fee to base denom. Send it to the community pool.")
-	}
+		k.Logger(ctx).With("fee", communityPool.String()).Debug("Sent fees to the community pool.")
 
-	// If the fee token is unknown or the swap is unsuccessful, emit event and return 
-	if baseDenomFee.Empty() {
 		err = uevent.EmitTypedEvent(ctx, &types.EventChargeFee{
-			Payer:              payer,
-			TakerFee:           takerFeeCoin.String(),
-			Beneficiary:        ValueFromPtr(beneficiary).String(),
-			BeneficiaryRevenue: "",
+			Payer:         payer,
+			TakerFee:      communityPool.String(),
+			CommunityPool: true,
 		})
 		if err != nil {
-			k.Logger(ctx).Error("Failed to emit event", "event", "EventChargeFee", "error", err)
+			return fmt.Errorf("emit event: %w", err)
```

Comment:
@mtsitrin @danwt rethinking about it, I think it probably makes more sense to not fail the charge fee flow if we can't emit events, so I do see more logic in previous work done by @keruch. 

wdyt?